### PR TITLE
[Feature] Support use catalog

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

we want to support "use catalog <catalog_name>".
If you send query "use catalog <catalog_name>" from mysql client, it will be parsed by mysql client to "use catalog" due to COM_INIT_DB protocol.  so you need to execute "**use 'catalog <catalog_name>'**".
The query from jdbc or bi tools will be parsed normally by COM_QUERY protocol.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 37 / 44 (84.09%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|3|5|60.00%|[88, 92]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|11|15|73.33%|[865, 866, 867, 877]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|7|8|87.50%|[2805]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseCatalogStmt.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -634,4 +637,8 @@ public void loop() {
             }
         }
     }
+
+    private boolean isChangeCatalog(String identifier) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:641]: I think you should implement this in one place.
1. to reduce duplicate code
2. to reduce unnecessary operation.

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 33 / 41 (80.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|3|6|50.00%|[90, 92, 96]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|11|15|73.33%|[865, 866, 867, 877]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|4|75.00%|[2800]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseDbStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/UseCatalogStmt.java|7|7|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -2150,18 +2151,24 @@ public ParseNode visitShowDatabasesStatement(StarRocksParser.ShowDatabasesStatem
...
+            return new UseDbStmt(null, parts.get(0));
         } else if (parts.size() == 2) {
-            return new UseStmt(parts.get(0), parts.get(1));
+            return new UseDbStmt(parts.get(0), parts.get(1));
         } else {
             throw new ParsingException("error catalog.database");
         }
     }
 
+    public ParseNode visitUseCatalog(StarRocksParser.UseCatalogContext context) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2166]: override？

**Review by DorianZheng — APPROVED** [review]
left a comment, rest LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**